### PR TITLE
ELPP-3693 Use tasks.create_ami from Jenkins

### PR DESCRIPTION
**Pull request by giorgiosironi**
Minimal extensions to make the task usable from an automated build that updates the AMI for the `containers` stack:

- fixes a bug with uncommitted files that are sometimes lost in the AMI
- put AMI ID output on a single line
- put most of the messages on stderr
- allow non-interactive confirmation through environment variable
- wait until AMI is `available`, it may be `pending` for a while
- allow a custom name to avoid clashes on `containers.2018-06-20` when running a build more than once in a day

### CODE REVIEW

File: scripts/clean-stack-for-ami.sh [hunk 1]
```diff
@@ -23,3 +22,7 @@ rm -rf \
     /etc/certificates/* \
     /root/events.log \
     /var/cache/salt/minion
+
+# commit memory buffers to disk to avoid partly written files
+# since AMIs are essentially snapshots of the root volume
+sync
```
Comment (lsh-0) [inline scripts/clean-stack-for-ami.sh]: nice one

File: src/buildercore/bakery.py [hunk 2]
```diff
@@ -10,18 +10,21 @@ def ami_name(stackname):
...
-def create_ami(stackname):
+def create_ami(stackname, name=None):
     "creates an AMI from the running stack"
     with core.stack_conn(stackname, username=config.BOOTSTRAP_USER):
-        bootstrap.prep_ec2_instance()
+        bootstrap.clean_stack_for_ami()
     ec2 = core.find_ec2_instances(stackname)[0]
     kwargs = {
-        'Name': ami_name(stackname),
+        'Name': ami_name(stackname) if name is None else name,
```
Comment (lsh-0) [inline src/buildercore/bakery.py]: `name or ami_name(stackname)` might be better

File: src/utils.py [hunk 3]
```diff
@@ -94,8 +95,12 @@ def uin(param, default=0xDEADBEEF):
 
 
 def confirm(message):
-    print(message)
-    print('press Enter to confirm (ctrl-c to quit)')
+    if config.BUILDER_NON_INTERACTIVE:
+        errcho('Non-interactive mode, confirming automatically')
```
Comment (lsh-0) [inline src/utils.py]: lacks context without the `message` being printed. I would suggest using the logger in non-interactive mode and have these as debug statements

Comment (giorgiosironi) [reply src/utils.py]: a `logger` should be a standard way to use `stderr`

Comment (lsh-0) [reply src/utils.py]: yeeeessss, I agree on principle. but to get the same functionality with a logger one has to create a special handler with a special formatter. it's not the first thought that comes to mind when you 'just need to print to stderr'.

Comment (giorgiosironi) [reply src/utils.py]: Able to add a logger, but will use `info` as a level so that the output is included in the Jenkins logs

**Pull request merged**